### PR TITLE
contrib: add basic database creator

### PR DESCRIPTION
**Pull request by mfrinnstrom**
## Hey, I just made a Pull Request!

I created a basic database creator for PostgreSQL and thought I could add that to the contrib folder if someone else is interested.

#### :heavy_check_mark: Checklist

- [X] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [X] Relevant documentation updated
- [X] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by andrewthauer** [comment]
@mfrinnstrom - It might be good to chime into #1598 as the discussion is very related.

File: contrib/database/basic-creator/database.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    );
+
+    return Promise.all(
+      plugins.map(plugin =>
+        createDatabaseIfNotExists(connection, plugin, logger),
+      ),
+    ).finally(() => connection.destroy());
+  }
+  logger.info('Not using PostgreSQL, skipping database checks...');
+  return Promise.resolve();
```
Comment (benjdlambert) [inline contrib/database/basic-creator/database.ts:58]: Don't think you need the `Promise.resolve` as it' defined as an `async` function.

**Review by benjdlambert — APPROVED** [review]

File: contrib/database/basic-creator/database.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getRootLogger } from '@backstage/backend-common';
+import { ConfigReader } from '@backstage/config';
+import knex, { Config } from 'knex';
+import { Logger } from 'winston';
+
+async function createDatabaseIfNotExists(
```
Comment (andrewthauer) [inline contrib/database/basic-creator/database.ts:21]: Considering this creates the db name based on the plugin, how bout renaming this to `createPluginDatabaseIfNotExists`? Same convention as below probably as well.

**Review by andrewthauer — COMMENTED** [review]
Just some suggestions. I think this has potential for being included into `@backstage/backend-common` as an optional database module extension. Again, recommend discussing further in #1598 .

**Comment by mfrinnstrom** [comment]
@andrewthauer & @benjdlambert thanks for you comments. I totally missed the discussion in #1598 but have heard it mentioned in Discord that you will look into this.

This was just a quick hack that I threw together that I thought could be useful for other people. I can see now that @lowjoel has started to work on this in #2674 and I think it is better to continue the work there then since it seems to have a bigger goal than this. I will add some comments to that instead.

**Pull request closed without merging**